elastic/kibana
# [Task Manager] Task Partitioning

Resolves https://github.com/elastic/kibana/issues/187698

## Summary


This PR does the following:
- Adds a new `partition` field to the task manager index
 - Assigns a partition to a task if there is not one when creating or updating

### Checklist



### To verify

New tasks
- Create a rule and verify that the newly created tasks have the `partition` field

Old tasks
- Checkout main and create a new rule, let it run
- Stop kibana
- Checkout this branch and restart kibana
- Verify that the old tasks get updated with the `partition` field

ex. the query I use to look at the ES query rule task
```
POST .kibana_task_manager*/_search
{
  "query": {
    "bool": {
      "filter": [
        {
          "term": {
            "task.taskType": {
              "value": "alerting:.es-query"
            }
          }
        }
      ]
    }
  }
}
```
